### ISSUE TITLE
Use JpegTurbo.jl to load .jpg images

### ISSUE THREAD

**Issue by lorenzoh** [FluxML/FastAI.jl#216]
JpegTurbo.jl can speed up image loading quite a bit, and the previous ImageIO.jl versions had severe performance regressions for 1.7, which should be fixed if JpegTurbo.jl is used.

**Comment by lorenzoh** [comment]
@johnnychen94 Will JpegTurbo.jl be used automatically if people use up-to-date FileIO.jl versions or do I need to change something explicitly? Would like users to avoid the performance regression introduced in 1.7 of the old jpg loader.

**Comment by johnnychen94** [comment]
TL;DR; add `ImageIO` to Project.toml without `using ImageIO` in `src/FastAI.jl` https://github.com/FluxML/FastAI.jl/pull/219

---

Let me try to explain the FileIO strategy a bit. FileIO maintains a registry for various IO backends. For instance, [there are four backends supporting JPEG format](https://github.com/JuliaIO/FileIO.jl/blob/21f435dc3ea63c4786e4cb894f855aef0692c056/src/registry.jl#L179-L186), i.e.,:

```julia
# Among the JPEG backends, JpegTurbo has the highest priority, QuartzImageIO is only available on macOS
add_format(
    format"JPEG",
    UInt8[0xff,0xd8,0xff],
    [".jpeg", ".jpg", ".JPG"],
    [idJpegTurbo],
    [idImageIO],
    [idQuartzImageIO, OSX],
    [idImageMagick]
) # 0xe1
```

Thus, when you try to do `load("file.jpg")`, FileIO will try to do `using JpegTurbo` and call JpegTurbo's load function. If this fails, then try the next one until it succeeds.

> the previous ImageIO.jl versions had severe performance regressions for 1.7

Before I wrote the JpegTurbo.jl, ImageMagick is the only IO backend for JPEG format, so when you see `load("file.jpg")` getting slow by a significant amount, it's because [ImageMagick is getting slow](https://github.com/JuliaIO/ImageMagick.jl/issues/208), which is now fixed in Julia 1.8.

**Comment by lorenzoh** [comment]
Thanks for the fix and the explanation!
